rosidl_parser/rosidl_parser/__init__.py

```diff
@@ -566,6 +566,17 @@ def parse_primitive_value_string(type_, value_string):
         for quote in ['"', "'"]:
             if value_string.startswith(quote) and value_string.endswith(quote):
                 value_string = value_string[1:-1]
+                if quote == '"':
+                    match = re.search(r'(?<!\\)"', value_string)
+                else:
+                    match = re.search(r"(?<!\\)'", value_string)
```

Comment:
Instead of repeating the same search twice I would suggest to insert the used `quote` into the pattern using string formatting.